### PR TITLE
feat: add mode option to artifact registry repository

### PR DESCRIPTION
**Pull request by FabrizioCafolla**

### CODE REVIEW

**Comment by FabrizioCafolla** [comment]
We waiting for the implementation on an [issue](https://github.com/hashicorp/terraform-provider-google/issues/17751) that allows using `customRepository`

**Comment by FabrizioCafolla** [comment]
This new feature for managing the docker custom registry require a version `hashicorp/google >= 5.26.0`

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,15 @@ Following semver, any non backwards compatible feature implies that the next rel
...
 ## [Unreleased]
 
+## [0.3.0] - 2024-05-02
+
+[Compare with previous version](https://github.com/sparkfabrik/terraform-google-gcp-artifact-registry/compare/0.2.1...0.3.0)
+
+### Changed
+
+- FEAT: Create an artifact registry repository in remote or virtual mode
+- FEAT: Upgrade `hashicorp/google` provider to `>= 5.26.0`
```
Comment (Monska85) [inline CHANGELOG.md]:
```suggestion
- FEATURE: Create an artifact registry repository in remote or virtual mode
- FEATURE: Upgrade `hashicorp/google` provider to `>= 5.26.0`
```

I would prefer to keep this convention, as described in the _header_ of this file.

File: examples/variables.tf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+variable "repositories" {
+  description = "List of Artifact Registry repositories to create."
+  type        = map(any)
+}
+
+variable "project_id" {
+  description = "The GCP project ID that hosts the Artifact Registry."
+}
```
Comment (Monska85) [inline examples/variables.tf]: Add the `type` to this variable (`string`).

**Review by Monska85 — CHANGES_REQUESTED** [review]

File: examples/variables.tf (hunk 2 above)
Comment (FabrizioCafolla) [reply examples/variables.tf]: Done

**Review by Monska85 — APPROVED** [review]
LGTM :+1:

**Pull request merged**